lib/spack/spack/detection/path.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""PATH based detection of software installed in the system"""
```

Comment:
I think the more-important theme in this module is that we are running executables: `executables_in_path` takes a `path_hints` parameter so is not strictly constrained to `PATH`.

Reply:
Modified docstring in 1dcf3a6